src/utils/page.js

```diff
@@ -38,6 +41,27 @@ export function getPage (win = window) {
   return detectedPageUrl
 }
 
+/**
+ * @return {string|undefined}
+ */
+export function getContextElements (contextSelectors, contextElementsLength, win = window) {
+  if (!contextSelectors || contextSelectors === '') {
+    return ''
+  } else {
+    var collectedElements = _collectElementsText(contextSelectors, win)
+    return base64UrlEncode(collectedElements).slice(0, contextElementsLength)
```

Comment:
This might produce an encoding that is considered invalid by some base64 parsers. Since base64 encodes a byte triplet to a byte quadruple, we could slice from 0 to contextElementsLength - contextElementsLength % 4 to produce a correct encoding.